instance/main.tf

```diff
@@ -1,72 +1,96 @@
 module "is_ebs_optimised" {
   source        = "../is_ebs_optimised"
-  instance_type = "${var.instance_type}"
+  instance_type = var.instance_type
 }
 
 resource "aws_instance" "instance" {
-  count                       = "${var.ebs_enabled ? var.instance_count : 0}"
-  ami                         = "${var.ami}"
-  instance_type               = "${var.instance_type}"
-  key_name                    = "${var.key_name}"
-  iam_instance_profile        = "${aws_iam_instance_profile.profile.id}"
-  vpc_security_group_ids      = ["${var.sgs}"]
-  subnet_id                   = "${element(var.subnets, count.index)}"
-  disable_api_termination     = "${var.termination_protection}"
-  ebs_optimized               = "${module.is_ebs_optimised.is_ebs_optimised}"
-  associate_public_ip_address = "${var.public_ip}"
-  user_data                   = "${element(var.user_data, count.index)}"
+  count                       = var.ebs_enabled ? var.instance_count : 0
```

Comment:
I think we can now use `dynamic` blocks to simplify the module and use a single `aws_instance` resource for both use cases (with and without ebs).